Updated platform exclusions

Alex-302 opened the pull request:
Added exclusions for `AdGuard for iOS` / `Safari`, `Content blocker for Android`
```
                "\\$domain=\/",
                ",$domain=\/",
                "\\$(.*,)?hls=",
                "\\$(.*,)?jsonprune="
```
`||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/` causes problems in `AdGuard for iOS / Safari` (see yesterday's thread in Safari chat)

`$domain=/` - in Safari, `if-domain` and `unless-domain` in compiled filters in JSON format dos not support regexps, only `*`

Details for basic `$domain` with regexp

<details>
<summary>Details</summary>

```
[16:07:58][LOG]:[$domain=/^i\[a-z\]*\.strmrdr\[a-z\]+\..*/]#%#//scriptlet('set-constant', 'adscfg.enabled', 'false') removed with removeRulePattern \$domain=/ in filter 13 for ios platform
[16:07:58][LOG]:||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/ removed with removeRulePattern \$domain=/ in filter 13 for ios platform
[16:07:58][LOG]:[$domain=/^i\[a-z\]*\.strmrdr\[a-z\]+\..*/]#%#//scriptlet('set-constant', 'adscfg.enabled', 'false') removed with removeRulePattern \$domain=/ in filter 13 for ext_safari platform
[16:07:58][LOG]:||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/ removed with removeRulePattern \$domain=/ in filter 13 for ext_safari platform
[16:07:59][LOG]:||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/ removed with removeRulePattern \$domain=/ in filter 13 for ext_android_cb platform
```

</details>



`hls=` and `jsonprune=` - supported only by the apps.

slavaleleka commented on custom_platforms.js:
```diff
@@ -223,7 +235,9 @@ module.exports = {
                 "\\$extension",
-                ",extension"
+                ",extension",
+                "\\$(.*,)?hls=",
```
why not more simple few separate patterns? just like for other modifiers
```
"\\$hls="
",hls=",
```

Alex-302 replied:
copy-pasta of existing modifiers

Alex-302 replied:
changed

AdamWr commented on custom_platforms.js:
```diff
@@ -82,7 +81,11 @@ module.exports = {
-                "\\$network"
+                "\\$network",
+                "\\$domain=\/",
+                ",$domain=\/",
```
Shouldn't it be: `,domain=\/`?

slavaleleka commented on custom_platforms.js:
@AdguardTeam/filters-maintainers a little bit off topic:
there are a lot of duplicates for different platforms. what do you think of using variables instead of string lists?

I mean something like this:
```
"removeRulePatterns": [
  "\\$extension",
  ",extension",
]
```
↓↓↓
```
const EXTENSION_MODIFIER_PATTERNS = [
  "\\$extension",
  ",extension",
];

"removeRulePatterns": [
  ...EXTENSION_MODIFIER_PATTERNS,
]
```

piquark6046 replied:
Then, what about creating javascript object of the patterns like `patterns.modifier.extension`?

Alex-302 replied:
it would be useful in hints and in directives. In conversations he once suggested.

slavaleleka commented:
@Alex-302 when these new patterns are approved, can you please also update the default compiler exclusions as well?

Alex-302 commented:
@slavaleleka 
> when these new patterns are approved, can you please also update the default compiler exclusions as well?

`exclusions.txt`? Why? It is not platforms related.

Alex-302 commented:
or this?
https://github.com/AdguardTeam/FiltersCompiler/blob/master/platforms.json

AdamWr commented on custom_platforms.js:
```diff
@@ -299,9 +316,12 @@ module.exports = {
-                ",$domain=\/",
-                "\\$(.*,)?hls=",
-                "\\$(.*,)?jsonprune="
+                ",domain=\/",
+                ",domain=\/",
```
It seems that ,domain=\/ is duplicated here.

AdamWr commented on custom_platforms.js:
```diff
@@ -236,8 +250,11 @@ module.exports = {
                 ",extension",
-                "\\$(.*,)?hls=",
-                "\\$(.*,)?jsonprune="
+                ",domain=\/",
```
Is it intentionally excluded from Chrome/Edge/Opera/Firefox extensions?
If so, shouldn't `\\$domain=\/` be also added?

Alex-302 replied:
Extensions for Chromium based browsers and FF will support this modifier later.

Alex-302 commented on custom_platforms.js:
```diff
@@ -177,9 +185,12 @@ module.exports = {
-                ",$domain=\/",
-                "\\$(.*,)?hls=",
-                "\\$(.*,)?jsonprune="
+                ",domain=\/",
+                ",domain=\/",
```
Ok. 
But you can make suggestions:)
![image](https://github.com/AdguardTeam/FiltersRegistry/assets/8361299/e72c4e24-f3d9-4199-8bf8-0f49cdb45e86)

AdamWr commented on custom_platforms.js:
```diff
@@ -320,7 +353,12 @@ module.exports = {
                 "$webrtc",
-                "#\\$#@media "
+                "#\\$#@media ",
+                ",domain=\/",
```
It looks like that something like this:
```adblock
test$xmlhttprequest,domain=/example\..*/
```
is already supported by uBlock Origin.
```suggestion
```

Alex-302 commented:
Result for these changes

<details>
<summary>Details</summary>

```
||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/ removed with removeRulePattern \$domain=/ in filter 13 for	ext_android_cb platform
||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_android_cb platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_android_cb platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_android_cb platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_android_cb platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_android_cb platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_android_cb platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_android_cb platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_android_cb platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ext_android_cb platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ext_android_cb platform

||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_chromium platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_chromium platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_chromium platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_chromium platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_chromium platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_chromium platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_chromium platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_chromium platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ext_chromium platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ext_chromium platform

||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_edge platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_edge platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_edge platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_edge platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_edge platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_edge platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_edge platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_edge platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ext_edge platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ext_edge platform

||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_ff platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_ff platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_ff platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_ff platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_ff platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_ff platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_ff platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_ff platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ext_ff platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ext_ff platform

||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_opera platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_opera platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_opera platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_opera platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_opera platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_opera platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_opera platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_opera platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ext_opera platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ext_opera platform

||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/ removed with removeRulePattern \$domain=/ in filter 13 for	ext_safari platform
||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_safari platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_safari platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_safari platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ext_safari platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_safari platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ext_safari platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_safari platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ext_safari platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ext_safari platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ext_safari platform

||video.twimg.com/ext_tw_video/*/*.m3u8$domain=/^i[a-z]*\.strmrdr[a-z]+\..*/ removed with removeRulePattern \$domain=/ in filter 13 for	ios platform
||turbopages.org^$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ios platform
||yandex.*/games/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ios platform
||yandex.*/q/$xmlhttprequest,jsonprune=\$..[direct\, "rtb"] removed with removeRulePattern ,jsonprune= in filter 1 for	ios platform
||drive2.ru/*/*/$xmlhttprequest,jsonprune=\$..[rtb\, direct] removed with removeRulePattern ,jsonprune= in filter 1 for	ios platform
||irecommend.ru/*/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ios platform
||ura.news/*/$xmlhttprequest,jsonprune=\$.data.*.attributes removed with removeRulePattern ,jsonprune= in filter 1 for	ios platform
||pubads.g.doubleclick.net/ondemand/hls/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=10play.com.au removed with removeRulePattern \$hls= in filter 2 for	ios platform
||dai.google.com/ondemand/hls/content/*.m3u8$hls=/redirector\.googlevideo\.com\/videoplayback[\s\S]*?dclk_video_ads/,domain=sbs.com.au removed with removeRulePattern \$hls= in filter 2 for	ios platform
||pluto.tv/*/session.json$jsonprune=\$..[adBreak\, adBreaks] removed with removeRulePattern \$jsonprune= in filter 2 for	ios platform
||pluto.tv/*/session.json$jsonprune=\$.adBreakLocations.* removed with removeRulePattern \$jsonprune= in filter 2 for	ios platform

```

</details>

AdamWr commented on custom_platforms.js:
```diff
@@ -223,7 +248,11 @@ module.exports = {
                 "important,protobuf",
                 "\\$extension",
-                ",extension"
+                ",extension",
```
It seems that `",domain=\/",` was accidentally removed from exclusions for Opera extension - https://github.com/AdguardTeam/FiltersRegistry/pull/806/commits/710e969aa8d916708d35398af27f675c83bcb4a9#diff-6d9b3e53da2e741716b096e56da5b27d6fb4daefcfce8e03d5a2f565a447da0fL253
```suggestion
                ",extension",
                ",domain=\/",
```

Alex-302 replied:
But this is Chromium based Opera. It was accidentally _added_ :)
